spec/rspec/mocks/any_instance_spec.rb

```diff
@@ -1213,6 +1230,17 @@ def dup(_)
         end
       end
 
+      context "on a grandchild class" do
+        it "stubs a method not implemented in grandchild" do
+          allow_any_instance_of(::AnyInstanceSpec::ChildClass).to receive(:foo).and_return :result
+          expect(::AnyInstanceSpec::ChildClass.new.foo).to eq :result
+        end
+
+        it "stubbed method doesn't get restored to grandchild" do
```

Comment:
Also, this test has a slight problem.  It only tests that the method is not restored if it runs _after_ the test above it, but since we use random ordering the order is not guaranteed (and we don't want order-dependent tests, anyway!).

Instead it should be written to do the any instance stubbing, restore things, and then make the expectation.  You can use [`reset_all`](https://github.com/rspec/rspec-mocks/blob/b94e242ebbc51081fc2245e37c574f5d6d02b7f8/spec/spec_helper.rb#L39-L41) to restore things.